src/goto-programs/remove_calls_nobody.cpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************\
+
+Module: Remove calls to functions without a body
+
+Author: Daniel Poetzl
+
+\*******************************************************************/
+
+/// \file
+/// Remove calls to functions without a body
+
+#include "remove_calls_nobody.h"
+
+void remove_calls_nobodyt::remove_call_nobody(
+  goto_programt &dest,
+  goto_programt::targett target,
+  const exprt &lhs,
+  const exprt::operandst &arguments)
+{
+  assert(target->is_function_call());
+  assert(!dest.empty());
+
+  goto_programt tmp;
+
+  // evaluate function arguments -- they might have
+  // pointer dereferencing or the like
+  forall_expr(it, arguments)
+  {
+    goto_programt::targett t=tmp.add_instruction();
+    t->make_other(code_expressiont(*it));
```

Comment:
An other expression just amounts to evaluating the expression. For example, for a call ``func(a[i])`` we insert an evaluation of ``a[i]`` so we can still detect out of bounds errors on the array access. The intention here is not to soundly model everything the function might do with a pointer argument but just to "simulate" the program evaluating the arguments to the call before making the call.